[CI] DatafeedJobsIT testDatafeedTimingStats_DatafeedRecreated failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #11428 / ubuntu-2404-aarch64_checkpart4_platform-support-arm](https://gradle-enterprise.elastic.co/s/fdltlsdj2ojty)
- [elasticsearch-periodic-platform-support #11428 / almalinux-8-aarch64_checkpart4_platform-support-arm](https://gradle-enterprise.elastic.co/s/wcs6jkhxtpl7o)
- [elasticsearch-periodic-platform-support #11412 / windows-2022_checkpart4_platform-support-windows](https://gradle-enterprise.elastic.co/s/7efllbruynkhm)
- [elasticsearch-pull-request #103677 / part-4](https://gradle-enterprise.elastic.co/s/lifrx7cm6dxyo)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:ml:qa:native-multi-node-tests:javaRestTest" --tests "org.elasticsearch.xpack.ml.integration.DatafeedJobsIT.testDatafeedTimingStats_DatafeedRecreated" -Dtests.seed=99F4095A4DB62A13 -Dtests.locale=ak-Latn-GH -Dtests.timezone=America/Yellowknife -Druntime.java=25
```

**Applicable branches:**
9.2

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.ml.integration.DatafeedJobsIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testDatafeedTimingStats_DatafeedRecreated'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <0L>
     but: was <3L>
```

**Issue Reasons:**
- [9.2] 2 consecutive failures in test testDatafeedTimingStats_DatafeedRecreated
- [9.2] 4 failures in test testDatafeedTimingStats_DatafeedRecreated (2.1% fail rate in 193 executions)
- [9.2] 2 failures in pipeline elasticsearch-periodic-platform-support (33.3% fail rate in 6 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -402,9 +402,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.optimizer.rules.logical.HoistRemoteEnrichTopNTests
   method: testTopNSortExpressionWithinRemoteEnrichAliasing
   issue: https://github.com/elastic/elasticsearch/issues/136957
-- class: org.elasticsearch.xpack.ml.integration.DatafeedJobsIT
-  method: testDatafeedTimingStats_DatafeedRecreated
-  issue: https://github.com/elastic/elasticsearch/issues/137207
 - class: org.elasticsearch.xpack.inference.DefaultEndPointsIT
   method: testMultipleInferencesTriggeringDownloadAndDeploy
   issue: https://github.com/elastic/elasticsearch/issues/117208

--- a/x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsIT.java
+++ b/x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsIT.java
@@ -250,7 +250,7 @@ public class DatafeedJobsIT extends MlNativeAutodetectIntegTestCase {
 
             putDatafeed(datafeedConfig);
             // Datafeed did not do anything yet, hence search_count is equal to 0.
-            assertDatafeedStats(datafeedId, DatafeedState.STOPPED, job.getId(), equalTo(0L));
+            assertBusy(() -> assertDatafeedStats(datafeedId, DatafeedState.STOPPED, job.getId(), equalTo(0L)));
             startDatafeed(datafeedId, 0L, now.toEpochMilli());
 
             // First, wait for data processing to complete
@@ -263,7 +263,14 @@ public class DatafeedJobsIT extends MlNativeAutodetectIntegTestCase {
                 TimeUnit.SECONDS
             );
 
-            deleteDatafeed(datafeedId);
+            DeleteDatafeedAction.Request request = new DeleteDatafeedAction.Request(datafeedId);
+            assertBusy(() -> {
+                try {
+                    client().execute(DeleteDatafeedAction.INSTANCE, request).actionGet();
+                } catch (Exception e) {
+                    throw new AssertionError("Datafeed could not be deleted", e);
+                }
+            });
             waitUntilJobIsClosed(job.getId());
         };
 

